src/Npgsql/NpgsqlSchema.cs

```diff
@@ -16,33 +18,38 @@ namespace Npgsql
     static class NpgsqlSchema
     {
         public static DataTable GetSchema(NpgsqlConnection conn, string? collectionName, string?[]? restrictions)
+            => GetSchemaAsync(conn, collectionName, restrictions, false).GetAwaiter().GetResult();
+
+        public static ValueTask<DataTable> GetSchemaAsync(NpgsqlConnection conn, string? collectionName, string?[]? restrictions, bool async = true, CancellationToken cancellationToken = default)
         {
             if (collectionName is null)
                 throw new ArgumentNullException(nameof(collectionName));
             if (collectionName.Length == 0)
                 throw new ArgumentException("Collection name cannot be empty.", nameof(collectionName));
 
+            cancellationToken.ThrowIfCancellationRequested();
+
             return collectionName.ToUpperInvariant() switch
             {
-                "METADATACOLLECTIONS"   => GetMetaDataCollections(),
-                "RESTRICTIONS"          => GetRestrictions(),
-                "DATASOURCEINFORMATION" => GetDataSourceInformation(conn),
-                "DATATYPES"             => GetDataTypes(conn),
-                "RESERVEDWORDS"         => GetReservedWords(),
+                "METADATACOLLECTIONS"   => new ValueTask<DataTable>(GetMetaDataCollections()),
+                "RESTRICTIONS"          => new ValueTask<DataTable>(GetRestrictions()),
+                "DATASOURCEINFORMATION" => new ValueTask<DataTable>(GetDataSourceInformation(conn)),
+                "DATATYPES"             => new ValueTask<DataTable>(GetDataTypes(conn)),
+                "RESERVEDWORDS"         => new ValueTask<DataTable>(GetReservedWords()),
                 // custom collections for npgsql
-                "DATABASES"             => GetDatabases(conn, restrictions),
-                "SCHEMATA"              => GetSchemata(conn, restrictions),
-                "TABLES"                => GetTables(conn, restrictions),
-                "COLUMNS"               => GetColumns(conn, restrictions),
-                "VIEWS"                 => GetViews(conn, restrictions),
-                "USERS"                 => GetUsers(conn, restrictions),
-                "INDEXES"               => GetIndexes(conn, restrictions),
-                "INDEXCOLUMNS"          => GetIndexColumns(conn, restrictions),
-                "CONSTRAINTS"           => GetConstraints(conn, restrictions, collectionName),
-                "PRIMARYKEY"            => GetConstraints(conn, restrictions, collectionName),
-                "UNIQUEKEYS"            => GetConstraints(conn, restrictions, collectionName),
-                "FOREIGNKEYS"           => GetConstraints(conn, restrictions, collectionName),
-                "CONSTRAINTCOLUMNS"     => GetConstraintColumns(conn, restrictions),
+                "DATABASES"             => GetDatabasesAsync(conn, restrictions, async),
```

Comment:
The standard naming for methods which accept the async flag is to not have the Async suffix (as they can be both)